packages/plugin-constraints/sources/tauModule.ts

```diff
@@ -0,0 +1,68 @@
+import {Project} from '@berry/core';
+import pl        from 'tau-prolog';
+
+const {is_atom} = pl.type;
+
+function prependGoals(
+    thread: pl.type.Thread, point: pl.type.State,
+    goals: pl.type.Term<number, string>[]): void {
+  thread.prepend(goals.map(
+      goal => new pl.type.State(
+          point.goal.replace(goal), point.substitution, point)));
+}
+
+const projects = new WeakMap<pl.type.Session, Project>();
+
+function getProject(thread: pl.type.Thread): Project {
+  const project = projects.get(thread.session);
+
+  if (project == null) {
+    throw new Error(`No project registered for the active session`);
+  }
+
+  return project;
+}
+
+const tauModule = new pl.type.Module(
```

Comment:
The constructor has a side effect, right?

Reply:
Yeah, the constructor registers the module in the tau-prolog package

https://github.com/jariazavalverde/tau-prolog/blob/4855fec296488e37e07698b7b716c5004ec4a9a2/modules/core.js#L1275